backend/fb.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Twin - A Tiny Window System
+ * Copyright (c) 2024 National Cheng Kung University, Taiwan
+ * All rights reserved.
+ */
+
+#include <fcntl.h>
+#include <linux/fb.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+#include "twin_fb.h"
+#include "twin_private.h"
+
+static void _twin_fb_put_begin(twin_coord_t left,
+                               twin_coord_t top,
+                               twin_coord_t right,
+                               twin_coord_t bottom,
+                               void *closure)
+{
+    twin_fb_t *tx = closure;
+    tx->width = right - left;
+    tx->height = bottom - top;
+    tx->image_y = top;
+}
+
+static void _twin_fb_put_span(twin_coord_t left,
+                              twin_coord_t top,
+                              twin_coord_t right,
+                              twin_argb32_t *pixels,
+                              void *closure)
+{
+    twin_fb_t *tx = closure;
+    twin_coord_t ix;
+    twin_coord_t iy = top;
+
+    for (ix = left; ix < right; ix++) {
+        twin_argb32_t pixel = *pixels++;
+
+        if (tx->depth == 16)
+            pixel = twin_argb32_to_rgb16(pixel);
+
+        tx->pixels[iy * tx->screen->width + ix] = pixel;
+    }
+    if ((top + 1 - tx->image_y) == tx->height) {
+        /* Update the pixels to the framebuffer */
+        memcpy(tx->fb_raw, tx->pixels,
+               sizeof(uint32_t) * tx->width * tx->height);
+    }
+}
+
+static bool twin_fb_read_events(int maybe_unused file,
+                                twin_file_op_t maybe_unused ops,
+                                void *closure)
+{
+    /* TODO: Implement events handling for mouse, keyboard, etc. */
+    return true;
+}
+
+static bool twin_fb_work(void *closure)
+{
+    twin_fb_t *tx = closure;
+
+    if (twin_screen_damaged(tx->screen))
+        twin_fb_update(tx);
+    return true;
+}
+
+twin_fb_t *twin_fb_create(char *fb_path)
+{
+    /* Allocate TWIN framebuffer object */
+    twin_fb_t *tx = malloc(sizeof(twin_fb_t));
+    if (!tx) {
+        fprintf(stderr, "Error in malloc.\n");
+        exit(2);
```

Comment:
Never call `exit` in the library, otherwise users fail to manipulate the control flow.